src/portable/synopsys/dwc2/dwc2_esp32.h

```diff
@@ -111,6 +123,31 @@ TU_ATTR_ALWAYS_INLINE static inline void dwc2_phy_update(dwc2_regs_t* dwc2, uint
   // maybe usb_utmi_hal_disable()
 }
 
+#ifdef DWC2_ENABLE_MEM_CACHE
+TU_ATTR_ALWAYS_INLINE static inline void dwc2_dcd_sync_cache_to_memory(void *addr, size_t size) {
+  int flags = ESP_CACHE_MSYNC_FLAG_DIR_C2M | ESP_CACHE_MSYNC_FLAG_UNALIGNED;
+  if (addr != NULL && size) {
+    ESP_EARLY_LOGV("ESP32_DWC", "cache to mem sync, addr 0x%"PRIx32", size %d", (uintptr_t)addr, size);
+    esp_err_t ret = esp_cache_msync(addr, size, flags);
+    assert(ret == ESP_OK);
+  }
+}
+
+TU_ATTR_ALWAYS_INLINE static inline void dwc2_dcd_sync_memory_to_cache(void *addr, size_t size) {
+  int flags = ESP_CACHE_MSYNC_FLAG_DIR_M2C;
+  if (addr != NULL && size) {
+    ESP_EARLY_LOGV("ESP32_DWC", "mem to cache sync, addr 0x%"PRIx32", size %d (%s)", (uintptr_t)addr, size);
```

Comment:
Not tested but this should be cleaner

```suggestion
    ESP_EARLY_LOGV("ESP32_DWC", "mem to cache sync, addr %"PRIxPTR", size %d (%s)", addr, size);
```

Reply:
I removed debug line from here at all.